mitodl/mit-learn
# Maintain API instructor order

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/5899

### Description (What does it do?)
Creates a new `RunInstructorRelationship` model with a `position` attribute.
Updates the `learning_resources.etl.loaders.load_instructors` function to assign `position` values based on the order received from the API.


### How can this be tested?
- Copy `MITX_ONLINE*` env settings from RC
- On the main branch, if you don't already have them, import some courses from mitxonline:
  ```python
  ./manage.py backpopulate_mitxonline_data
  ```
- If/when you do have instructors, switch to this branch and restart containers.  A series of migrations should complete successfully.
- Go to http://open.odl.local:8062/search/?q=%22Moral%20Problems%20and%20the%20Good%20Life%22 and open the course drawer for the 1st result.  Note that the order of instructors does not match what is returned by the API for that course, which is:
  ```
                "instructors": [
                    {
                        "name": "Kieran Setiya",
                        "description": ""
                    },
                    {
                        "name": "Caspar Hare",
                        "description": ""
                    },
                    {
                        "name": "Tamar Schapiro",
                        "description": ""
                    },
                    {
                        "name": "David Balcarras",
                        "description": ""
                    }
                ]
  ```
  ![Screenshot 2024-11-01 142542](https://github.com/user-attachments/assets/c0da59c7-d4fc-4037-982f-f73dd7de5b78)

- Run `./manage.py backpopulate_mitxonline_data`
- Refresh the course drawer above.  Now the instructor order should match the api's order.
  
  ![Screenshot 2024-11-01 143537](https://github.com/user-attachments/assets/d9a82113-8fe9-47d7-9328-b79e8d5d5b39)
